src/platform/qpg6100/BLEManagerImpl.cpp

```diff
@@ -120,53 +132,37 @@ CHIP_ERROR BLEManagerImpl::_SetFastAdvertisingEnabled(bool val)
 
 CHIP_ERROR BLEManagerImpl::_GetDeviceName(char * buf, size_t bufSize)
 {
-    CHIP_ERROR err = CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+    CHIP_ERROR err;
 
-    // TODO
-    // err = qvCHIP_BleGetDeviceName(buf, bufSize);
-    goto exit;
+    err = qvCHIP_BleGetDeviceName(buf, bufSize);
 
-exit:
     return err;
 }
 
 CHIP_ERROR BLEManagerImpl::_SetDeviceName(const char * devName)
 {
-    CHIP_ERROR err = CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+    CHIP_ERROR err;
 
-    // TODO
-    // err = qvCHIP_BleSetDeviceName(devName);
-    goto exit;
+    if (mServiceMode == ConnectivityManager::kCHIPoBLEServiceMode_NotSupported)
+    {
+        return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+    }
+    if (devName != NULL && devName[0] != 0)
+    {
+        err = qvCHIP_BleSetDeviceName(devName);
```

Comment:
should this err (if it happens) be returned? Alternatively if this is not success, should we log it?

it seems like there is some double-return happening below.

Reply:
Correct - Remnant return from 'goto exit:' removal